src/node_file.cc

```diff
@@ -940,16 +940,16 @@ static void InternalModuleReadJSON(const FunctionCallbackInfo<Value>& args) {
   char** ppos = &pos[0];
 
   while (p < pe) {
-    char c = *p++;
-    if (c == '\\' && p < pe && *p == '"') p++;
-    if (c != '"') continue;
+    const char c = *p++;
+    if (c == '\\' && *p == '"') continue;
```

Comment:
Read past end - it's possible for `p == pe` because of the `*p++` on the previous line.